gocourse06/homework.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"sync"
+	"time"
+)
+
+type Animal struct {
+	Name   string
+	Health int
+	Hunger int
+	State  string
+}
+
+type Cage struct {
+	ID     int
+	IsOpen bool
+}
+
+type Feeder struct {
+	ID     int
+	IsFull bool
+}
+
+func main() {
+	var wg sync.WaitGroup
+	processData(&wg)
+}
+
+func processData(wg *sync.WaitGroup) {
+	animals := makeAnimals()
+	animalsChannel := make(chan Animal)
+	wg.Add(len(animals))
+	for _, animal := range animals {
+		go func(animal Animal) {
```

Comment:
Ось це вже не потрібно, бо в нас Go 1.23.
```suggestion
		go func() {
```

Ось тут пояснено https://go.dev/blog/go1.22#language-changes

Reply:
Те саме для всіх інших горутин.